### DISCUSSION TITLE
Debug extension disabled (break point ignored)

### DISCUSSION THREAD

**Discussion by frog-o** [coder/code-server#5413] [Q&A]
First I know I am playing around with experimental software  setting so I am ask for head aches.  So this is manly to talk about what I found out.  I been trying to use esm modules with typescript using tsx or ts-node. I never like having a separate directory for compiling.  I been able to run both of these without issue until now after I upgrade code server from the dpkg release page (since the Debian packages I can get by run  apt-update is not new enough, I originally had a lot of extension  complaining the vscode api was to old.) even thought that fix the api to old problem it seem to have broken the debugger.

After a bunch of play around trying to get the debugger to work in my ava test ,( it is ignoring all my break point) I discover that JavaScript debug was disabled.  It tells me it is disable because it will only  run on code-server for the desktop.  I think this is what cause everything not to work.

Is it.?
What are people doing about this?
I currently looking into other extensions and I wondering if anyone else is having this problem?
and what they are using?

I currently have this in my launch.json
```
"type": "node",
            "request": "launch",
            "name": "Debug AVA test file",
            "args": [ "--inspect,--loader=ts-node/esm,--experimental-specifier-resolution=node"],
            "runtimeExecutable": "${workspaceFolder}/node_modules/ava/entrypoints/cli.mjs",
            "program": "${workspaceFolder}/test/src/utils.ts",
            "console": "integratedTerminal", // optional

and this in package.json

"ava": {
    "extensions": {
      "ts": "module"
    },
    "nodeArguments": [
      "--experimental-specifier-resolution=node",
      "--loader=tsx",
      "--inspect"
    ],
    "files": [
      "test/src/utils.ts"
    ]
  }


            
```

**Comment by jsjoeio** [comment]
Can you provide basic steps to reproduce this? Developing extensions in code-server has had some trickiness to it in the past. That could be related to this. When you develop it in VS Code, do you run into this problem?

**Comment by frog-o** [comment]
First thanks for replying and just to be clear, i am not making an
extension but a node program.
It hard to write a step by step when you're trouble shooting, one of the
things of trouble shooting is to see if others are having the problem.  If
others don't have the visual studio code extension disable in code-server
then i think it probably a weird bug.  My guess is All   you have to do is
create a node program and put break points in it. In reality it could be as
simple as install the latest code server and look at disable extension and
I think you see the JavaScript debugger disabled. I went as far back on the
4.0  releases as i could, the oldest 4.0 realize would start code-server, i
even tried the newest realize with an old version (as far back as i could
go) of the JavaScript debugger , it is still disabled. Maybe code-server
debugging  never worked. I go back and forth between code-server and visual
studio code so that much is hard to tell. I going look into debugger using
node built in debugger next.  But trouble shooting takes time and if no one
else is having a problem not really worth much trying to talk about it, at
least not till  I get it figure out.

**Reply by jsjoeio** [reply]
Ahh okay. So in summary, debugging via break points in Node programs in code-server isn't working for you?

If you can provide some basic steps to reproduce and what you're seeing, i can try locally!

**Comment by frog-o** [comment]
I have a lot of things i wonder if other people have problems with but the
biggest is JavaScript debugger disabled for other people.

1. Install code-server
2. Click on add extension icon button
3. Click the filter button
4. See if the JavaScript debugger is in the list

If enable what version of code-server are you using and with what version
of JavaScript debugger.

I pretty sure code-server can't debug anything related to JavaScript
Without a job script debugger extension. I would think this would be hard
to do.  The weird thing is some functionality works, if i run the node
debugger in the console it will stop on the first line and i can click the
run button in the debug panel and it will run but ignore the break points.


By the way if i run my program in vs code it works without a problem it is
definitely a code-server a problem.

I think for some debugging it uses a built in chrome debugger. I think it
probably only works those type of apps. I got a lot more test to do and no
time to test.

**Reply by jsjoeio** [reply]
Oddly enough, it says "Reload Required" which makes me think it's not working with 4.5.1 (cc @code-asher any ideas?)

<img width="1712" alt="image" src="https://user-images.githubusercontent.com/3806031/183711934-6ce89fec-8a46-4413-911e-585a397f2cef.png">

Okay so it doesn't work if you choose a browser like Chrome
<img width="1712" alt="image" src="https://user-images.githubusercontent.com/3806031/183712187-7914ed8e-11fc-4aab-adfc-4b1539732055.png">

But it does work if you choose Node.js 🎉

<img width="1712" alt="image" src="https://user-images.githubusercontent.com/3806031/183712405-c28307e6-535e-4721-91a8-f12ab1921ced.png">

**Reply by code-asher** [reply]
Hmm not sure about the "reload required".  I remember we had that issue and patched it but I forgot what happened after that.  https://github.com/coder/code-server/issues/4600

Maybe we lost the patch?

**Reply by jsjoeio** [reply]
Ah interesting. Well maybe we lost it? If we add it back, I'd be happy to write an e2e test for it.

**Comment by frog-o** [comment]
Mine said disabled.  I don't think it the same bug. It is interesting yours
said reload required.  Maybe  it's related to the problem I have with
self-sainted certificates not working. I think i could originally debug a
JavaScript program but not a program run through ava testing library.
I think this probably a multi threading issue but  i haven't felt  up to
testing much and wondering if other had problem with the debug disable
issue.  I done a seach and most issue come from source map being turned off
or forgetting the --loader tsx or --loader /register/ts-node(or whatever it
is flag) i tried them both. I couldn't find anyone that said it is
disabled.
